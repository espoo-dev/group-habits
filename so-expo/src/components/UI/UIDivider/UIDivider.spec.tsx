import { render, screen } from '@testing-library/react-native';
import { UIDivider } from './index';

function setup() {
  return render(<UIDivider />)
}

describe('src > components > UI > UIDivider', () => {
  it(`match snapshot`, () => {
    const tree = setup().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly`, () => {
    setup()

    const text = screen.getByText(/ou/i)

    expect(text).toBeOnTheScreen()
  })
});
