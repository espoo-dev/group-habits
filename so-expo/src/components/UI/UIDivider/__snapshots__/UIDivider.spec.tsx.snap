// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src > components > UI > UIDivider match snapshot 1`] = `
<View
  style={
    [
      {
        "position": "relative",
      },
      {
        "display": "flex",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    aria-hidden={true}
    style={
      [
        {
          "position": "absolute",
        },
        {
          "top": 0,
        },
        {
          "right": 0,
        },
        {
          "bottom": 0,
        },
        {
          "left": 0,
        },
        {
          "display": "flex",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "width": "100%",
          },
          {
            "borderTopWidth": 1,
          },
          {
            "borderBottomColor": "#d1d5db",
            "borderLeftColor": "#d1d5db",
            "borderRightColor": "#d1d5db",
            "borderTopColor": "#d1d5db",
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "position": "relative",
        },
        {
          "display": "flex",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "paddingLeft": 8,
            "paddingRight": 8,
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "color": "#6b7280",
          },
        ]
      }
    >
      Ou
    </Text>
  </View>
</View>
`;
